### ISSUE TITLE
We need a codebook for Student Data

### ISSUE THREAD

**Issue by jhconning** [jhconning/StudentData#1]
Following Partha's suggestion. I'll setup a very simple codebook on the wiki page some time soon, that I hope others can help improve over time. 

> ---------- Forwarded message ----------
> From: Partha Deb <partha.deb@hunter.cuny.edu>
> Date: Mon, Oct 12, 2015 at 10:49 AM
> Subject: Re: Fwd: The files are ready for download
> To:
> I forgot to do this a few days ago, but there is a corrected and merged course-student panel dataset > at Dropbox.  As I suggested, it will be very important to go through it immediately to check for errors, > things we don't understand, etc.  Please -codebook- all the variables, and that will give a sense of  where to look.

**Comment by harryfujiwara** [comment]
Okay - am working on this now

**Comment by pdebpdeb** [comment]
In fact, I would not use -compact- . Do the full -codebook- because the typical values in each variable are insightful - esp. to detect errors.

**Comment by jhconning** [comment]
Great I'm mentioning @mbaker21231 here to remind him to [watch](https://help.github.com/articles/watching-repositories/) this repository and hence get email notification on issue tracking.. I'll remind Partha and Moayad and Devra to share their github account names and  become [watchers](https://github.com/jhconning/StudentData/watchers) to follow as well. 

  I think this will make it easier to move the project forward than disparate email threads.

**Comment by mbaker21231** [comment]
Everyone - I'm now watching. Thanks for reminding me Jonathan!

**Comment by pdebpdeb** [comment]
I'm pdebpdeb at github

On 10/14/2015 2:30 PM, Jonathan Conning wrote:
>
> Great I'm mentioning @mbaker21231 <https://github.com/mbaker21231> 
> here to remind him to watch 
> <https://help.github.com/articles/watching-repositories/> this 
> repository and hence get email notification on issue tracking.. I'll 
> remind Partha and Moayad and Devra to share their github account names 
> and become watchers 
> <https://github.com/jhconning/StudentData/watchers> to follow as well.
>
> I think this will make it easier to move the project forward than 
> disparate email threads.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jhconning/StudentData/issues/1#issuecomment-148146468>.
>

-- 
Partha Deb
Professor of Economics
Director of Graduate Studies
Hunter College
ph:  (212) 772-5435
fax: (212) 772-5398
http://econ.hunter.cuny.edu/parthadeb/

Emancipate yourselves from mental slavery
None but ourselves can free our minds.
	- Bob Marley

**Comment by harryfujiwara** [comment]
Given that a student has retaken MATH150, here is the distribution of grades and the attempt number.  Note that all of the students in Attempt #1 are students that took MATH150 and repeated it at least once.  Thus this does not include students who failed the first time and never retook it

![image](https://cloud.githubusercontent.com/assets/15126372/10495920/532e5e76-728c-11e5-822e-13eaced34fcf.png)

**Comment by pdebpdeb** [comment]
This cool.  Could you please add to this table the row percents, i.e., percent of 2 attempts with each grade, percent of 3 attempts with each grade, ...

Also, it would be interesting to compare with other - lower math classes. 100, 150, ...

**Comment by pdebpdeb** [comment]
I meant 100, 125, ...

**Comment by dgolbe** [comment]
FWIW, when i click on the "view it on Github" link I don't see this post.

Am I reading this correctly? 1688 students retook Math 150 at least 
once.  Of those, 1034 got an F on the first retake.  All 1688 students 
retook it a 2nd time (so 3 total attempts);  497 failed a third time.  
It doesn't seem right that every single person who repeated once 
repeated again.

**Comment by harryfujiwara** [comment]
Given that a student has retaken MATH150, here is the distribution of grades by the attempt number.

Note that all of the students in Attempt 1 are students that took MATH150 and repeated it at least once. Thus this does not include students who failed the first time and never retook it

![image](https://cloud.githubusercontent.com/assets/15126372/10496206/9046cf54-728d-11e5-88b6-6f450caba78f.png)

**Comment by harryfujiwara** [comment]
No... 1,688 students retook MATH 150.  The FIRST time they took it, 1,034 of them failed, 197 got D's, etc.  When they retook it (attempt #2), 497 failed AGAIN... 202 got As.  The reason that the totals in column 1 and 2 are the same is because we are only considering the subset of students who retook MATH 150.  All students who retake a class take it (at least) twice

**Comment by harryfujiwara** [comment]
Professor Deb's request (percentage distribution, by attempt number)

![image](https://cloud.githubusercontent.com/assets/15126372/10496383/8d759dc2-728e-11e5-9396-94894d01ffd8.png)

**Comment by jhconning** [comment]
Very nice.. Can you place your stata code and output on the dropbox folder?  Also can you report tables for Eco 300.

**Comment by harryfujiwara** [comment]
It's in a folder in the dropbox called "Repeated classes"

Here it is for ECO30000:

![image](https://cloud.githubusercontent.com/assets/15126372/10497125/475247ba-7292-11e5-9891-d274e54b4b4f.png)

**Comment by harryfujiwara** [comment]
And MATH 100:

![image](https://cloud.githubusercontent.com/assets/15126372/10497330/3e6759b4-7293-11e5-8a1d-c44e396d1113.png)

**Comment by harryfujiwara** [comment]
Professor Deb was wondering what relationship MATH 100 grades have with ENG 120 grades.  Here's my preliminary results:

           |                         eng120
   math100 |         A          B          C          D          F |     Total
-----------+-------------------------------------------------------+----------
         A |       694        226         78         14          9 |     1,021 
         B |       420        243         67         17         18 |       765 
         C |       351        205         59         13         12 |       640 
         D |       191        155         54         12         13 |       425 
         F |       146        147         49         16         94 |       452 
-----------+-------------------------------------------------------+----------
     Total |     1,802        976        307         72        146 |     3,303 

As Professor Deb suspected, kids who do well in one tend to do well in the other.  And clearly English 120 is much easier than Math 100...

I saved my do file in the same folder
